IngressMaxFields.user.js

```diff
@@ -69,66 +71,79 @@ function wrapper() {
 		});
 		return c;
 	}
+	self.inBounds = function(portal) {
+		if (window.plugin.drawTools && window.plugin.drawTools.drawnItems.getLayers().length) {
+			return self.portalInDrawnItems(portal);
+		} else {
+			return self.portalInScreen(portal);
+		}
+	}
+	self.genStr = function genStr(p,x) {
+		var href = "https://www.ingress.com/intel?ll=" + p._latlng.lat + "," + p._latlng.lng + "&z=17&pll=" + p._latlng.lat + "," + p._latlng.lng;
+		var str = p.options.data.title;
+		str = str.replace(/\"/g, "\\\"");
+		str = str.replace(";"," ");
+		if (window.plugin.keys && (typeof window.portals[x] !== "undefined")) {
+			var keyCount = window.plugin.keys.keys[x] || 0;
+			str = str + ";" +href +";" + keyCount;
+		} else {
+			str = str + ";" + href;
+		}
+		return str;
+	}
 
-	self.gen = function gen() {
-			var o = [];
-			var antal = 0;
-			var tooMany = 0;
-
-			if (window.plugin.drawTools && window.plugin.drawTools.drawnItems.getLayers().length) {
-				var inBounds = function(portal) {
-					return self.portalInDrawnItems(portal);
-				}
-				var string = "Portal selection based on drawTools boundaries.";
-			} else {
-				var inBounds = function(portal) {
-					return self.portalInScreen(portal);
-				}
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Function 'checkPortals' has a complexity of 6.](https://www.codacy.com/app/stefan-nygren/IITC-Ingress-Maxfields-Exporter/file/1697783007/issues/source?bid=3039347&fileBranchId=3039566#l95)